raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -76,4 +81,56 @@ library ChannelManagerLibrary {
 
         return channelAddress;
     }
+
+    /// @notice deleteChannel(address) to remove a channel after it's been settled
+    /// @dev Remove channel after it's been settled
+    /// @param channelAddress (address) address of the channel to be closed
+    /// @param partner (address) address of the partner
+    function deleteChannel(Data storage self, address partner, address channelAddress) private {
+
+        address[] ourChannels = self.nodeChannels[msg.sender];
+        address[] partnerChannels = self.nodeChannels[partner];
+
+        // remove element from sender
+        for (uint i = 0; i < ourChannels.length; ++i) {
+            if (ourChannels[i] == channelAddress) {
+                ourChannels[i] = ourChannels[ourChannels.length -1];
```

Comment:
Also @czepluch @schmir these are the current rules regarding arrays and the way they are handled in solidity:

Local variables are by default in storage.
Function arguments are in memory.

Arrays in Storage are copied
Arrays in memory are referenced